Publish and update maven cache

lkiesow opened the pull request:
We have a lot of automated build downloading dependencies from Maven Central and mvn.opencast.org over and over again. This takes time and occasionally causes build failures.

This patch adds a workflow building up a repository cache once a week and making it publically available, so that you can just get most of the dependencies in just a single tarball. This should make things faster and easier.

### Your pull request should…

* [x] have a concise title
* [x] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

lkiesow committed:
Publish and update maven cache

We have a lot of automated build downloading dependencies from Maven
Central and mvn.opencast.org over and over again. This takes time and
occasionally causes build failures.

This patch adds a workflow building up a repository cache once a week
and making it publically available, so that you can just get most of the
dependencies in just a single tarball. This should make things faster
and easier.

gregorydlogan commented on .github/workflows/build-maven-cache.yml:
```diff
@@ -0,0 +1,76 @@
+
+      - name: build opencast
+        run: |
+          mvn clean install -Pdev \
```
Does it make sense to use `dependency:go-offline` instead?  Then we're not packaging the resulting OC jars as well, which we (probably) don't actually want?

lkiesow replied:
I didn't know about `dependency:go-offline`. That sounds like a good idea.
Would the command you are suggesting be:
```
          mvn dependency:go-offline -Pdev \
            --batch-mode \
            -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn \
            -Dhttp.keepAlive=false \
            -Dmaven.wagon.http.pool=false \
            -Dmaven.wagon.httpconnectionManager.ttlSeconds=120
```

gregorydlogan replied:
I would normally have left the `-Pdev` off, but realistically it's not going to matter either way.

lkiesow replied:
I used `-Pdev` instead of `-Pnone` so that maven will fetch Karaf, its plugins and dependencies which I think it otherwise wouldn't do, would it?

gregorydlogan replied:
No idea, but including the dev stuff makes sense to me.

lkiesow replied:
Unfortunately, `mvn dependency:go-offline` fails. I've worked around that by now by using the previous command but removing the Opencast JARs from the cache before creating the tarball.

gregorydlogan commented on .github/workflows/build-maven-cache.yml:
```diff
@@ -0,0 +1,76 @@
+
+      - name: upload assets
+        run: |
+          s3cmd put -P dot-m2.tar.xz s3://opencast-maven-cache/
```
This workflow doesn't explicitly say which branch(es) it runs on, but they're all going to write to the same place.  I'm assuming this only runs against the main branch when it's unspecified?

Also, we're building this cache, should we use it in the builds?

lkiesow replied:
Scheduled workflows always run on the default branch. But even using that, it should provide most dependencies for release branches.

I'd like to use this in builds in the long run. But I thought it makes sense to get this uploading first, and then we can play with using this for different builds.

gregorydlogan replied:
Fair.  Builds in a followup PR then.

lkiesow committed:
Simplify s3cmd configuration

lkiesow committed:
Remove Opencast JARs from Maven cache

This patch makes sure the Opencast assets itself will not end up in the
maven cache pushed to S3 to ensure that all Opencast related modules are
freshly built.

lkiesow commented:
This should be ready to go now
